src/schemas/produceMessageSchema.ts

```diff
@@ -58,5 +64,55 @@ export const PRODUCE_MESSAGE_SCHEMA: JSONSchema = {
       },
       required: ["key", "value"],
     },
+    schemaInfo: {
+      type: "object",
+      properties: {
+        // TODO: add Schema Registry URI?
+        schema_version: {
+          type: "integer",
+          description: "The version number of the schema",
+        },
+        subject: {
+          type: "string",
+          description: "The subject of the schema",
+        },
+        subject_name_strategy: {
+          type: "string",
+          description: "The subject name strategy",
+          enum: ["TOPIC_NAME", "RECORD_NAME", "TOPIC_RECORD_NAME"],
+        },
+      },
+      required: ["schema_version", "subject", "subject_name_strategy"],
+    },
   },
 };
+
+export interface ProduceMessage {
+  key?: string;
+  key_schema?: SchemaInfo;
+  value?: string;
+  value_schema?: SchemaInfo;
+  headers?: MessageHeader[];
+  partition_id?: number;
+  timestamp?: number;
+}
+
+export enum SubjectNameStrategy {
+  TOPIC_NAME = "TOPIC_NAME",
+  RECORD_NAME = "RECORD_NAME",
+  TOPIC_RECORD_NAME = "TOPIC_RECORD_NAME",
+}
+
+export interface SchemaInfo {
+  // TODO: add Schema Registry URI?
+  schema_version: number;
+  subject: string;
+  subject_name_strategy: SubjectNameStrategy | undefined;
+  type?: "BINARY" | "JSON" | "STRING" | undefined;
```

Comment:
Not used for validation in the spec, just internally while preparing the produce request body